sdk/eventhub/Azure.Messaging.EventHubs/tests/Infrastructure/EventProcessorManager.cs

```diff
@@ -147,6 +147,132 @@ public Task StopAllAsync()
                 .Select(eventProcessor => eventProcessor.StopAsync()));
         }
 
+        /// <summary>
+        ///   Waits until the partition load distribution is stabilized.  Throws a <see cref="TimeoutException"/> if
+        ///   the load takes too long to stabilize.
+        /// </summary>
+        ///
+        /// <returns>A task to be resolved on when the operation has completed.</returns>
+        ///
+        public async Task WaitStabilization()
+        {
+            var stabilizedStatusAchieved = false;
+            var consecutiveStabilizedStatus = 0;
+            List<PartitionOwnership> previousActiveOwnership = null;
+
+            var startTime = DateTimeOffset.UtcNow;
+
+            while (!stabilizedStatusAchieved)
+            {
+                // Give up if it takes more than 1 minute.
+
+                var elapsedTime = DateTimeOffset.UtcNow.Subtract(startTime);
```

Comment:
I'd recommend setting a `CancellationTokenSource` at the beginning with your desired timeout and then calling `Token.ThrowIfCancellationRequested` here, rather than doing the clock management yourself.